flink-runtime/src/test/java/org/apache/flink/runtime/io/network/util/DeserializationUtils.java

```diff
@@ -36,18 +40,25 @@
 	 *
 	 * @param records records to be deserialized
 	 * @param deserializer the record deserializer
+	 * @param mustBeFullRecords if set, fails if the deserialized records contain partial records
 	 * @return the number of full deserialized records
 	 */
 	public static int deserializeRecords(
 			ArrayDeque<SerializationTestType> records,
-			RecordDeserializer<SerializationTestType> deserializer) throws Exception {
+			RecordDeserializer<SerializationTestType> deserializer,
+			boolean mustBeFullRecords) throws Exception {
 		int deserializedRecords = 0;
 
 		while (!records.isEmpty()) {
 			SerializationTestType expected = records.poll();
 			SerializationTestType actual = expected.getClass().newInstance();
 
-			if (deserializer.getNextRecord(actual).isFullRecord()) {
+			RecordDeserializer.DeserializationResult deserializationResult =
+				deserializer.getNextRecord(actual);
+			if (mustBeFullRecords) {
+				assertThat(deserializationResult, hasProperty("fullRecord", equalTo(true)));
```

Comment:
why not `assertTrue(deserializationResult.isFullRecord())`?

Reply:
(Slightly) better context:
```
java.lang.AssertionError: 
Expected: hasProperty("fullRecord", <true>)
     but: property 'fullRecord' was <false>

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at org.apache.flink.runtime.io.network.util.DeserializationUtils.deserializeRecords(DeserializationUtils.java:61)
```
vs.
```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at org.apache.flink.runtime.io.network.util.DeserializationUtils.deserializeRecords(DeserializationUtils.java:61)
```

This sometimes allows faster debugging because for some things you don't have to click into the source to know what actually failed.